### ISSUE TITLE
ARN attribute is null for non-existant ECR loaded as data resource

### ISSUE THREAD

**Issue by JonathanHolvey** [terraform-providers/terraform-provider-aws#9947]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.3
provider.aws v2.26.0

### Affected Resource

* aws_ecr_repository

### Terraform Configuration Files

```hcl
provider "aws" {
  region = "ap-southeast-2"
}

data "aws_ecr_repository" "main" {
  name = "example/repo-name"
}

data "aws_iam_policy_document" "ecr_role_policy" {
  statement {
    actions = [
      "ecr:*"
    ]
    effect = "Allow"
    resources = [
        data.aws_ecr_repository.main.arn,
    ]
  }
}

```

### Debug output

```
Error: Null value found in list

  on main.tf line 15, in data "aws_iam_policy_document" "ecr_role_policy":
  15:     resources = [
  16:         data.aws_ecr_repository.main.arn,
  17:     ]

Null values are not allowed for this attribute value.
```

### Expected Behaviour

Terraform should display an error when the ECR cannot be found in the specified provider account.

### Actual Behaviour

The ARN resolves to `Null` and Terraform complains that null values are not allowed in the policy document `resources` attribute value.

### Steps to Reproduce

1. `terraform plan` using the config above

**Comment by tedivm** [comment]
I can into this today- it also provides null output for the repository url instead of giving a proper error.

**Comment by bflad** [comment]
Hi folks 👋  The fix for this data source to properly return an error (https://github.com/terraform-providers/terraform-provider-aws/pull/10520) was recently merged and will release with version 3.0.0 of the Terraform AWS Provider, likely in about a week.
